core/app/components/Memberships/MembersList.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+
+import { Avatar, AvatarFallback, AvatarImage } from "ui/avatar";
+
+import type { MembersListProps, TargetId } from "./types";
+import { RemoveMemberButton } from "./RemoveMemberButton";
+import { RoleSelect } from "./RoleSelect";
+
+export const MembersList = <T extends TargetId>({
+	members,
+	setRole,
+	removeMember,
+	targetId,
+	readOnly,
+}: MembersListProps<T>) => {
+	return (
+		<>
+			{members.map((user) => (
+				<div key={user.id} className="flex items-center gap-4">
```

Comment:
these member lists look really good! really nice you can just edit the membership inline
this is super nitpicky so feel free to ignore, but i think they'd look slightly nicer if the badge was right aligned and slightly smaller! see this and the next comment

```suggestion
				<div key={user.id} className="flex items-center justify-between gap-4">
```